external/fv3fit/fv3fit/data/tfdataset.py

```diff
@@ -16,8 +17,7 @@
 
 def stack(ds: xr.Dataset, unstacked_dims: Sequence[str]):
     stack_dims = [dim for dim in ds.dims if dim not in unstacked_dims]
-    unstacked_dims = [dim for dim in ds.dims if dim in unstacked_dims]
-    unstacked_dims.sort()  # needed to always get [x, y, z] dimensions
```

Comment:
was this `.sort()` causing an issue for your feature? Or did you just remove since it's unnecessary?

Reply:
It was causing issues. The tests don't care, but it broke my models for production because my post-evaluation scripts expect specific dimension orders, and they were no longer followed when I added `.sort` in a previous commit.

Reply:
Seems fine, but can you jog my memory why you had commented in before that a specific order was needed?

Reply:
Yes... this code is used is many places, so some care required here IMO.